solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -261,22 +260,23 @@ public String getMessage() {
     }
   }
 
-  private static HttpClient createHttpClient(
-      SolrCore core,
-      String httpBasicAuthUser,
-      String httpBasicAuthPassword,
-      boolean useCompression) {
+  private Http2SolrClient createHttpClient(
+      SolrCore core, String httpBasicAuthUser, String httpBasicAuthPassword, String leaderBaseUrl) {
     final ModifiableSolrParams httpClientParams = new ModifiableSolrParams();
```

Comment:
This is now unused.  I'd hope that authentication stuff would be handled in some other way (e.g. via configuration / initialization in UpdateShardHandler creating the recoveryOnlyHttpClient) so that internal Solr code here can merely get this pre-configured client.  That client should also already have an InstrumentedHttpListenerFactory too; no?

Reply:
I did some digging here.  IndexFetcher can be configured with httpBasicAuthUser and httpBasicAuthPassword in its solrconfig.xml based configuration however this isn't document, and seems rather ancient / legacy to me.  Let's just drop this.  Not sure it's even worth mentioning in upgrade notes.

Reply:
(this is still pending action)

Reply:
Do you see my point that `httpClientParams` is now unused?  At a minimum, it needs to be removed now.

With your insights on httpBasicAuthUser and httpBasicAuthPassword configuration, a comment explaining *why* we can't simply assume the PKI one would be helpful to whoever reads this in the future.